errors/errors.go

```diff
@@ -0,0 +1,147 @@
+package errors
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ErrUnknown is an unknown error type
+const ErrUnknown = Error("unknown error")
+
+// Error is a string that implements error
+// this will have interesting side effects of having constant errors
+type Error string
+
+// Error returns error message
+func (e Error) Error() string {
+	return string(e)
+}
+
+// New returns a new error with message passed.
+// if args are passed, we will format the message with args
+// Example:
+// New("some error") returns error with exact string passed
+// New("some error: %v", "some context") returns error with message "some error: some context"
+func New(msg string, args ...interface{}) error {
```

Comment:
Can we enforce here the format? Meaning:
```
err := New("some error", "some context")
```
And it is the `New` function that formats it:
```
...
if len(args) > 0 {
		return Error(fmt.Sprintf("%s: %s", msg, args...))
}
```
That way we have the standard built in.

Reply:
We can enforce it but leaves with multiple functions like std library. Thinking having a single function can be easier.

Reply:
But we are trying to standardize our usage of errors right? 
In addition, looking at the `msg string` argument, doesn't tell me if it is just a string or a format string, so either we make it clear in the arg name and doc or we enforce it, IMHO.

Reply:
Doc says it already. Not sure what could be the name of arg can be. Do you have a suggestion for arg name?

Reply:
Will update the arg to format as per the std lib